jib-cli/src/test/java/com/google/cloud/tools/jib/cli/jar/JarProcessorTest.java

```diff
@@ -18,30 +18,260 @@
 
 import static com.google.common.truth.Truth.assertThat;
 
+import com.google.cloud.tools.jib.api.buildplan.AbsoluteUnixPath;
+import com.google.cloud.tools.jib.api.buildplan.FileEntriesLayer;
+import com.google.cloud.tools.jib.api.buildplan.FileEntry;
 import com.google.cloud.tools.jib.cli.jar.JarProcessor.JarType;
+import com.google.common.collect.ImmutableList;
 import com.google.common.io.Resources;
 import java.io.IOException;
 import java.net.URISyntaxException;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
 
 public class JarProcessorTest {
 
-  private static final String SPRING_BOOT_RESOURCE_DIR = "jar/springboot/springboot_sample.jar";
-  private static final String STANDARD_RESOURCE_DIR = "jar/standard/standardJar.jar";
+  private static final String SPRING_BOOT_JAR = "jar/springboot/springboot_sample.jar";
+  private static final String STANDARD_JAR_WITH_CLASS_PATH_MANIFEST =
+      "jar/standard/standardJarWithClassPath.jar";
+  private static final String STANDARD_JAR_WITHOUT_CLASS_PATH_MANIFEST =
+      "jar/standard/standardJarWithoutClassPath.jar";
+  private static final String STANDARD_JAR_WITH_ONLY_CLASSES =
+      "jar/standard/standardJarWithOnlyClasses.jar";
+  private static final String STANDARD_JAR_EMPTY = "jar/standard/emptyStandardJar.jar";
+
+  @Rule public final TemporaryFolder temporaryFolder = new TemporaryFolder();
 
   @Test
   public void testDetermineJarType_springBoot() throws IOException, URISyntaxException {
-    Path springBootJar = Paths.get(Resources.getResource(SPRING_BOOT_RESOURCE_DIR).toURI());
+    Path springBootJar = Paths.get(Resources.getResource(SPRING_BOOT_JAR).toURI());
     JarType jarType = JarProcessor.determineJarType(springBootJar);
     assertThat(jarType).isEqualTo(JarType.SPRING_BOOT);
   }
 
   @Test
   public void testDetermineJarType_standard() throws IOException, URISyntaxException {
-    Path standardJar = Paths.get(Resources.getResource(STANDARD_RESOURCE_DIR).toURI());
+    Path standardJar = Paths.get(Resources.getResource(STANDARD_JAR_EMPTY).toURI());
     JarType jarType = JarProcessor.determineJarType(standardJar);
     assertThat(jarType).isEqualTo(JarType.STANDARD);
   }
+
+  @Test
+  public void testExplodeMode_standard_emptyJar() throws IOException, URISyntaxException {
+    Path standardJar = Paths.get(Resources.getResource(STANDARD_JAR_EMPTY).toURI());
+    Path destDir = temporaryFolder.newFolder().toPath();
+    List<FileEntriesLayer> layers = JarProcessor.explodeStandardJar(standardJar, destDir);
+
+    // Validate that only two layers are created.
```

Comment:
super nit: I think this comment is obvious enough to be removed.

Reply:
True, done.